File: frontend/lib/models/lost_and_found.dart
class LostAndFoundItem {
  String userId;
  String name;
  String lastSeenLocation;
  String imagePath;
  String description;
  String contactNumber;
  bool isLost;

  LostAndFoundItem({
    required this.userId,
    required this.name,
    required this.lastSeenLocation,
    required this.imagePath,
    required this.description,
    required this.contactNumber,
    required this.isLost,
  });

  factory LostAndFoundItem.fromJson(Map<String, dynamic> json) {
    return LostAndFoundItem(
      userId: json['user_id'],
      name: json['name'],
      lastSeenLocation: json['lastSeenLocation'],
      imagePath: json['imagePath'],
      description: json['description'],
      contactNumber: json['contactNumber'],
      isLost: json['isLost'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'user_id': userId,
      'name': name,
      'lastSeenLocation': lastSeenLocation,
      'imagePath': imagePath,
      'description': description,
      'contactNumber': contactNumber,
      'isLost': isLost,
    };
  }
}

File: backend/models/lost_and_found.js
import mongoose from "mongoose";

const lostAndFoundItemSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true },
    lastSeenLocation: { type: String, trim: true },
    imagePath: { type: String, default: null }, // Prevents undefined errors
    description: { type: String, trim: true },
    contactNumber: {
      type: String,
      required: true,
      trim: true,
      match: [/^\d{10}$/, "Please enter a valid 10-digit phone number"], // Enforces valid numbers
    },
    listerId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Student",
      required: false,
      index: true, // Faster queries on lost items per user
    },
    isLost: { type: Boolean, required: true },
  },
  { timestamps: true } // Auto adds createdAt & updatedAt
);

const LostAndFoundItem = mongoose.model(
  "LostAndFoundItem",
  lostAndFoundItemSchema
);
export default LostAndFoundItem;